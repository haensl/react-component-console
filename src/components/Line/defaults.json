{
  "classes": {
    "element": "Console-Line",
    "content": "Console-Line-content"
  }
}
