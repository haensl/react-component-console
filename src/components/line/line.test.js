import React from 'react';
import renderer from 'react-test-renderer';
import Line, { defaults } from './';

describe('Line', () => {
  describe('empty', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<Line />)
        .toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });
  });

  describe('ps1', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<Line ps1={{
        content: '~'
      }} />).toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });

    it('sets the tilde as content of the ps1 component', () => {
      expect(tree.children[0].children.includes('~')).toBe(true);
    });
  });

  describe('content', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<Line content="foo bar" />)
        .toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });

    it('adds "foo bar" as content', () => {
      expect(tree.children[0].children.includes('foo bar')).toBe(true);
    });
  });

  describe('classes', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<Line classes={{
        element: 'test-line',
        content: 'test-line-content'
      }} />).toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the test-line element class', () => {
      expect(tree.props.className).toEqual('test-line');
    });

    it('adds the test-line-content content class', () => {
      expect(tree.children[0].props.className).toEqual('test-line-content');
    });
  });

  describe('write', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<Line write={ true } cursor={{
        char: 't'
      }} />).toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });

    it('sets the char under the cursor correctly', () => {
      expect(tree.children[1].children.includes('t')).toBe(true);
    });
  });
});
