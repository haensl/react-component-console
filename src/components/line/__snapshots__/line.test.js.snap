// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line classes renders as expected 1`] = `
<pre
  className="test-line"
>
  <span
    className="test-line-content"
  />
</pre>
`;

exports[`Line content renders as expected 1`] = `
<pre
  className="console-line"
>
  <span
    className="console-line-content"
  >
    foo bar
  </span>
</pre>
`;

exports[`Line empty renders as expected 1`] = `
<pre
  className="console-line"
>
  <span
    className="console-line-content"
  />
</pre>
`;

exports[`Line ps1 renders as expected 1`] = `
<pre
  className="console-line"
>
  <span
    className="console-ps1"
  >
    ~
  </span>
  <span
    className="console-line-content"
  />
</pre>
`;

exports[`Line write renders as expected 1`] = `
<pre
  className="console-line"
>
  <span
    className="console-line-content"
  />
  <span
    className="console-cursor console-cursor--write"
  >
    t
  </span>
</pre>
`;
